# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/windows/ia2/ia2AccessibleRelation.cpp
# Commit: 8d31c0604efe
# Full Hash: 8d31c0604efef2015995df32766d38dd4dd45ca2
# Author: James Teh <jteh@mozilla.com>
# Date: 2022-08-04 04:34:13
# Regressor Bug: 1781536
# File Overlap Count: 1
# Description:
#   Bug 1781536 part 1: Support base Accessible in Relation. r=morgan
#   
#   Previously, Relation::Next returned a LocalAccessible.
#   Now, it returns an Accessible so that it can support RemoteAccessible as well.
#   Relation::LocalNext has been added as a convenience to return a LocalAccessible when the caller is certain that only LocalAccessible is relevant.
# ==============================================================================

diff -r af7f9589f48c -r 8d31c0604efe accessible/windows/ia2/ia2AccessibleRelation.cpp
--- a/accessible/windows/ia2/ia2AccessibleRelation.cpp	Thu Aug 04 02:47:34 2022 +0300
+++ b/accessible/windows/ia2/ia2AccessibleRelation.cpp	Wed Aug 03 23:58:50 2022 +0000
@@ -16,8 +16,10 @@
 
 ia2AccessibleRelation::ia2AccessibleRelation(RelationType aType, Relation* aRel)
     : mType(aType) {
-  LocalAccessible* target = nullptr;
-  while ((target = aRel->Next())) mTargets.AppendElement(target);
+  Accessible* target = nullptr;
+  while ((target = aRel->Next())) {
+    mTargets.AppendElement(MsaaAccessible::GetFrom(target));
+  }
 }
 
 // IUnknown
@@ -69,8 +71,7 @@
       !aTarget)
     return E_INVALIDARG;
 
-  RefPtr<IAccessible> target;
-  mTargets[aTargetIndex]->GetNativeInterface(getter_AddRefs(target));
+  RefPtr<IAccessible> target = mTargets[aTargetIndex];
   target.forget(aTarget);
 
   return S_OK;